azure-proto-core/GenericResourcesOperations.cs

```diff
@@ -0,0 +1,52 @@
+using Azure;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace azure_proto_core
+{
+    /// <summary>
+    /// Implementation for resources that implement the immutable resource pattern
+    /// </summary>
+    public abstract class GenericResourcesOperations<TResource, TOperations> : OperationsBase 
```

Comment:
Can this just be ResourceOperations?

Reply:
yeah Generic has other meaning when it comes to results from Arm vs results from an RP